src/test/run-make-fulldeps/coverage-reports/expected_show_coverage.conditions.txt

```diff
@@ -38,69 +38,53 @@
-   63|       |        // not executed if the first `then` condition is true), only the condition's code region is
-   64|       |        // marked with a count of 0 now.
-   65|      0|        else if countdown > 2 {
-   66|      0|            if countdown < 1 || countdown > 5 || countdown != 9 {
-   67|      0|                countdown = 0;
-   68|      0|            }
-   69|      0|            countdown -= 5;
-   70|       |        } else {
-   71|      0|            return;
-   72|       |        }
-   73|      1|    }
-   74|       |
-   75|      1|    let mut countdown = 0;
-   76|      1|    if true {
-   77|      1|        countdown = 1;
-   78|      1|    }
-   79|       |
-   80|      1|    let z = if countdown > 7 {
+   40|      1|        let mut countdown = 0;
+   41|      1|        if true {
+   42|      1|            countdown = 10;
+   43|      1|        }
+   44|       |
+   45|      1|        if countdown > 7 {
+   46|      1|            countdown -= 4;
+   47|      1|        }
+   48|      0|        else if countdown > 2 {
+   49|      0|            if countdown < 1 || countdown > 5 || countdown != 9 {
+   50|      0|                countdown = 0;
+   51|      0|            }
+   52|      0|            countdown -= 5;
+   53|       |        } else {
+   54|      0|            return;
+   55|       |        }
+   56|       |    } // Note: closing brace shows uncovered (vs. `0` for implicit else) because condition literal
+   57|       |      // `true` was const-evaluated. The compiler knows the `if` block will be executed.
+   58|       |
+   59|      1|    let mut countdown = 0;
+   60|      1|    if true {
+   61|      1|        countdown = 1;
+   62|      1|    }
+   63|       |
+   64|      1|    let z = if countdown > 7 {
                       ^0
-   81|      0|        countdown -= 4;
-   82|      1|    } else if countdown > 2 {
-   83|      0|        if countdown < 1 || countdown > 5 || countdown != 9 {
-   84|      0|            countdown = 0;
-   85|      0|        }
-   86|      0|        countdown -= 5;
-   87|       |    } else {
-   88|      1|        let should_be_reachable = countdown;
-   89|      1|        println!("reached");
-   90|      1|        return;
-   91|       |    };
-   92|       |
-   93|      0|    let w = if countdown > 7 {
-   94|      0|        countdown -= 4;
-   95|      0|    } else if countdown > 2 {
-   96|      0|        if countdown < 1 || countdown > 5 || countdown != 9 {
-   97|      0|            countdown = 0;
-   98|      0|        }
-   99|      0|        countdown -= 5;
-  100|       |    } else {
-  101|      0|        return;
-  102|       |    };
-  103|      1|}
+   65|      0|        countdown -= 4;
+   66|      1|    } else if countdown > 2 {
+   67|      0|        if countdown < 1 || countdown > 5 || countdown != 9 {
+   68|      0|            countdown = 0;
+   69|      0|        }
+   70|      0|        countdown -= 5;
+   71|       |    } else {
```

Comment:
ditto (and below)

Reply:
To me, these all look completely accurate. Here's the color version including these lines...

<img width="443" alt="Screen Shot 2020-12-01 at 10 16 19 PM" src="https://user-images.githubusercontent.com/3827298/100835739-ef1b2c80-3422-11eb-8e01-a2eb6f6cf47f.png">

* `countdown > 7` is checked (`false`)
* `countdown > 2` is checked (`false`)
* the `else` block is executed

Reply:
I see, I guess because there is an `else` block, all the information is still there. What confused me is the inconsistency with `if` expressions that have no `else` block. If we always included the span after the closing brace as part of the `else` block it would be more consistent and self-explanatory, in my opinion.

Reply:
I get your point, but that's not how the HIR spans set it up. (But that could be changed, as I did here for implicit else.) I'm not sure if your suggestion is best in the long run... or if there are other implications (e.g., with if-else-if-else.). Maybe.

Reply:
Ah so I think doing something like what you did for all if exprs (not just those that have no `else`) would give more consistent looking output, personally, and I think it can work fine with chained `else if`s.

Reply:
Yeah perhaps. I'm still concerned about unexpected side effects.